pkg/chunkenc/memchunk.go

```diff
@@ -1761,6 +1765,15 @@ func (e *entryBufferedIterator) StreamHash() uint64 { return e.pipeline.BaseLabe
 
 func (e *entryBufferedIterator) Next() bool {
 	for e.bufferedIterator.Next() {
+		// If this is a NoProcessingStreamPipeline, bypass processing
+		if e.isNoProcessingPipeline {
+			e.currLabels = e.pipeline.BaseLabels()
+			e.cur.Timestamp = time.Unix(0, e.currTs)
+			e.cur.Line = string(e.currLine)
+			e.cur.StructuredMetadata = logproto.FromLabelsToLabelAdapters(e.currStructuredMetadata)
```

Comment:
I suspect we could do this without allocation but I'm hesitant to make that change because of the risk associated and I believe this PR on it's own would be a significant improvement